src/VideoStreaming/VideoReceiver.cc

```diff
@@ -966,6 +968,61 @@ VideoReceiver::_keyframeWatch(GstPad* pad, GstPadProbeInfo* info, gpointer user_
 }
 #endif
 
+void VideoReceiver::takeSnapshot() {
+#if defined(QGC_GST_STREAMING)
+    qDebug() << "Taking Snapshot";
+    if (!_videoSink) {
+        return;
+    }
+
+    qDebug() << "Trying to take a snapshot";
+    GstSample *videobuffer = nullptr;
+    GstCaps *caps;
+    gint width, height;
+    GstMapInfo map;
+
+    g_object_get(G_OBJECT(_videoSink), "last-sample", &videobuffer, NULL);
+    if (!videobuffer) {
+        qDebug() << "No video buffer, exiting";
+        return;
+    }
+
+    caps = gst_sample_get_caps(videobuffer);
+    if (!caps) {
+        qDebug() << "Cant get caps from video buffer, exiting";
+        return;
+    }
+
+    GstStructure *s = gst_caps_get_structure(caps, 0);
+    gboolean res = gst_structure_get_int (s, "width", &width)
+                && gst_structure_get_int (s, "height", &height);
+    if (!res) {
+        qDebug() << "Cant get width or height from caps";
+        return;
+    }
+
+    GstBuffer *snapbuffer = gst_sample_get_buffer(videobuffer);
+    gst_buffer_map (snapbuffer, &map, GST_MAP_READ);
+
+    uchar* bufferData = reinterpret_cast<uchar*>(map.data);
+    QImage::Format imageFormat = QImage::Format_RGB32;
+    // QImage::InvertMode invertMode = QImage::InvertRgb;
```

Comment:
stale? leave an explanation if commented code should stay around

Reply:
IMO this code should check buffer format and if it is not RGB32 (RGBA as qmlglsink advertises) then simply fail - potential crash place (low probability, but...).

Reply:
not stale, the picture is being saved with the wrong colors and I'm experimenting.

Reply:
There are should be some kind 'inverted' RGB32 image format - most likely problem is related to color component order - what you have might be ABGR. Try to look for some BGR32 or InvertedRGB32 - something like.